src/main/java/backend/Logic.java

```diff
@@ -157,32 +160,57 @@ public void refresh() {
+            Date lastSelfUpdate = new Date(0);
+            boolean isUpdatedByOthers = false;
+            boolean isUpdatedBySelf = false;
             for (TurboIssueEvent event : currentMetadata.getEvents()) {
                 if (!event.getActor().getLogin().equalsIgnoreCase(currentUser)
                         && event.getDate().after(lastNonSelfUpdate)) {
                     lastNonSelfUpdate = event.getDate();
+                    isUpdatedByOthers = true;
+                } else if (event.getActor().getLogin().equalsIgnoreCase(currentUser)
+                        && event.getDate().after(lastSelfUpdate)){
+                    lastSelfUpdate = event.getDate();
+                    isUpdatedBySelf = true;
                 }
             }
             for (Comment comment : currentMetadata.getComments()) {
                 if (!comment.getUser().getLogin().equalsIgnoreCase(currentUser)
                         && comment.getCreatedAt().after(lastNonSelfUpdate)) {
                     lastNonSelfUpdate = comment.getCreatedAt();
+                    isUpdatedByOthers = true;
+                } else if (comment.getUser().getLogin().equalsIgnoreCase(currentUser)
+                        && comment.getCreatedAt().after(lastSelfUpdate)){
+                    lastSelfUpdate = comment.getCreatedAt();
+                    isUpdatedBySelf = true;
                 }
             }
 
             entry.setValue(new IssueMetadata(currentMetadata,
                     LocalDateTime.ofInstant(lastNonSelfUpdate.toInstant(), ZoneId.systemDefault()),
-                    calculateNonSelfCommentCount(currentMetadata.getComments(), currentUser)
+                    LocalDateTime.ofInstant(lastSelfUpdate.toInstant(), ZoneId.systemDefault()),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, IssueType.OTHER_UPDATED),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, IssueType.SELF_UPDATED),
+                    isUpdatedByOthers,
+                    isUpdatedBySelf
             ));
         }
         return metadata;
     }
 
-    private int calculateNonSelfCommentCount(List<Comment> comments, String currentUser) {
+    private int calculateCommentCount(List<Comment> comments, String currentUser, IssueType issueType) {
         int result = 0;
-        for (Comment comment : comments) {
-            if (!comment.getUser().getLogin().equalsIgnoreCase(currentUser)) {
-                result++;
+        if (issueType == IssueType.OTHER_UPDATED) {
+            for (Comment comment : comments) {
+                if (!comment.getUser().getLogin().equalsIgnoreCase(currentUser)) {
+                    result++;
+                }
+            }
+        } else if (issueType == IssueType.SELF_UPDATED){
+            for (Comment comment : comments) {
+                if (comment.getUser().getLogin().equalsIgnoreCase(currentUser)) {
+                    result++;
+                }
```

Comment:
The conditional logic should be factored out into a predicate so we don't have to repeat ourselves. This may also be formulated more clearly in terms of a `filter`, then `count`, but we can try the predicate first.